### ISSUE TITLE
UselessVariable: False positive for static variable

### ISSUE THREAD

**Issue by Levivb** [slevomat/coding-standard#502]
Hey,

The following code snippet reports a UselessVariable, while it's not useless:

```php
public function getThing(): string
    {
        static $result;

        if (is_string($result)) {
            return $result;
        }

        $result = someHeavyTask();

        return $result;
    }
```

The static part followed by a `return if static var is something` ensures some heavy calculation or task is not run more then necessary. The code sniffer unfortunately reports a SlevomatCodingStandard.Variables.UselessVariable.UselessVariable

Let me know if you need more information!

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/9978172758e90bc2355573e0b5d99062d87b14a3
